ui/curses.c

```diff
@@ -810,6 +986,11 @@ void mtr_curses_open(
     int bg_col = 0;
     int i;
 
+#ifdef WITH_BRAILLE_DISPLAY
+    // initialize all locale variables, before ncurses starts
+    setlocale(LC_ALL, "");
+#endif
```

Comment:
You said you had removed this...

Reply:
I removed `setenv("LOCALE", "", 1)` which was the hack.

If you look at the man page for locale you'll find...
> The argument category determines which parts of
> the program's current locale should be modified.
>
> LC_ALL                  - All of the locale

and
> If locale is an empty string, "", each part of the locale that should be
> modified is set according to the environment variables.

This is the way one enables locale support, using environment variables set by the user.